metron-platform/metron-hbase-server/src/main/java/org/apache/metron/hbase/coprocessor/EnrichmentCoprocessor.java

```diff
@@ -0,0 +1,198 @@
+  private GlobalConfigService globalConfigService;
+  private RegionCoprocessorEnvironment coprocessorEnv;
+  private Map<String, Object> globalConfig;
+
+  /**
+   * HBase requires a no-arg constructor.
+   */
+  public EnrichmentCoprocessor() {
+  }
+
+  /**
+   * Allow test dep injection.
+   */
+  public EnrichmentCoprocessor(CacheWriter<String, String> cacheWriter,
+      GlobalConfigService globalConfigService) {
+    this.cache = Caffeine.newBuilder().writer(cacheWriter).build();
+    this.globalConfigService = globalConfigService;
+  }
+
+  /**
+   * Allow test dep injection.
+   */
+  public EnrichmentCoprocessor(GlobalConfigService globalConfigService) {
+    this.globalConfigService = globalConfigService;
+  }
+
+  @Override
+  public void start(CoprocessorEnvironment ce) throws IOException {
+    LOG.info("Starting enrichment coprocessor");
+    if (ce instanceof RegionCoprocessorEnvironment) {
+      this.coprocessorEnv = (RegionCoprocessorEnvironment) ce;
+    } else {
+      throw new CoprocessorException("Enrichment coprocessor must be loaded on a table region.");
+    }
+    LOG.info("Checking if internal cache initialized");
+    if (null == this.cache) {
+      LOG.info("Cache null, initializing");
+      LOG.info("Getting global config from Zookeeper");
+      String zkUrl = getZookeeperUrl(this.coprocessorEnv.getConfiguration());
+      if (null == globalConfigService) {
+        globalConfigService = getGlobalConfigService(zkUrl);
+      }
+      globalConfig = globalConfigService.get();
+      Configuration config = this.coprocessorEnv.getConfiguration();
+      CacheWriter<String, String> cacheWriter = null;
+      try {
+        String hbaseTableProviderName = (String) globalConfig
+            .get(EnrichmentConfigurations.TABLE_PROVIDER);
+        String tableName = (String) globalConfig.get(EnrichmentConfigurations.TABLE_NAME);
+        String columnFamily = (String) globalConfig.get(EnrichmentConfigurations.COLUMN_FAMILY);
+        cacheWriter = new HBaseCacheWriter(config, TableProvider
+            .create(hbaseTableProviderName, HTableProvider::new), tableName, columnFamily,
+            COLUMN_QUALIFIER);
+      } catch (ClassNotFoundException | InstantiationException | InvocationTargetException | IllegalAccessException | NoSuchMethodException e) {
+        throw new IOException("Unable to instantiate cache writer", e);
+      }
+      this.cache = Caffeine.newBuilder().writer(cacheWriter).build();
+      LOG.info("Finished initializing cache");
+    }
+    LOG.info("Finished starting enrichment coprocessor");
+  }
+
+  private String getZookeeperUrl(Configuration config) {
+    String zkUrl = config.get(ZOOKEEPER_URL);
+    if (null == zkUrl) {
+      throw new IllegalStateException(
+          "Enrichment coprocessor requires property '" + ZOOKEEPER_URL
+              + "' to be provided at startup.");
+    }
+    return zkUrl;
+  }
+
+  private GlobalConfigService getGlobalConfigService(String zkUrl) {
```

Comment:
We have a `org.apache.metron.common.zookeeper.ZKConfigurationsCache` abstraction that does the same thing.  Does it make sense to reuse that here?  I wouldn't say it's a requirement because this is just a one time load but could be useful in the future if we do need to access the global config after a coprocessor is started.  Just something for you to consider.

Reply:
I had considered that, but I wanted to be careful because, as you mentioned, this is a one time load and there could be adverse side effects without explicitly changing it to handle changes to the configured parameters. The other thing is that the ZKConfigurationsCache abstraction is tightly coupled to the concept of our topology types, e.g. 
* ENRICHMENT
* PARSER
* INDEXING
* PROFILER

Any reference to the global config has to be through one of those configuration types. There is probably some refactoring that could be done there if we wanted, but it felt a bit ham-fisted to shoehorn that in for what amounts to about 10 fairly concise lines of code.

Reply:
As long as you are aware of that and made a conscious decision to not use it then it's fine with me.